publisher.go

```diff
@@ -0,0 +1,178 @@
+package main
+
+import (
+	"fmt"
+	"time"
+	"net/http"
+	"io/ioutil"
+	"log"
+	"github.com/jlaffaye/ftp"
+	"bytes"
+	"sync"
+	"io"
+)
+
+var lock = &sync.Mutex{}
+var wg = &sync.WaitGroup{}
+
+
+func processFiles(files <- chan string) {
+	defer wg.Done()
+	conn, err := connectToFtp()
+	if err != nil {
+		log.Fatal("Unable to connect to FTP server exiting - aborting")
+	}
+	defer conn.Logout()
```

Comment:
Shouldn't strictly need to worry about the `Logout` - the `Quit` will close the connection anyhow